java-compiler-testing/src/main/java/io/github/ascopes/jct/compilers/AlreadyUsedCompilerException.java

```diff
@@ -0,0 +1,34 @@
+package io.github.ascopes.jct.compilers;
```

Comment:
So I think we'll need the license header at the top of each of these files. You can copy-paste that from any other Java file, but Checkstyle will fail without it being here! 

```suggestion
/*
 * Copyright (C) 2022 Ashley Scopes
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.github.ascopes.jct.compilers;
```